src/Functions/JSONPath/Parsers/ParserJSONPathMemberSquareBracketAccess.cpp

```diff
@@ -0,0 +1,51 @@
+#include "ParserJSONPathMemberSquareBracketAccess.h"
+#include <memory>
+#include <Functions/JSONPath/ASTs/ASTJSONPathMemberAccess.h>
+#include <IO/ReadBufferFromMemory.h>
+#include <IO/ReadHelpers.h>
+#include <Parsers/ASTIdentifier.h>
+#include <Parsers/ExpressionElementParsers.h>
+
+namespace DB
+{
+bool ParserJSONPathMemberSquareBracketAccess::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
+{
+    if (pos->type != TokenType::OpeningSquareBracket)
+        return false;
+    ++pos;
+    ASTPtr member_name;
+    if (pos->type == TokenType::BareWord || pos->type == TokenType::QuotedIdentifier)
+    {
+        ParserIdentifier name_p;
+        if (!name_p.parse(pos, member_name, expected))
+            return false;
+    }
+    else if (pos->type == TokenType::StringLiteral)
+    {
+        try
+        {
+            ReadBufferFromMemory in(pos->begin, pos->size());
+            String name;
+            readQuotedStringWithSQLStyle(name, in);
+            member_name = std::make_shared<ASTIdentifier>(name);
+            ++pos;
+        }
+        catch (const Exception &)
+        {
+            return false;
```

Comment:
a comment is needed, with explanation why we need try catch.

Reply:
It's for early debug. I remove it.